executor/fktest/foreign_key_test.go

```diff
@@ -417,6 +417,41 @@ func TestForeignKeyCheckAndLock(t *testing.T) {
 		wg.Wait()
 		tk.MustQuery("select id, name from t1 order by name").Check(testkit.Rows("1 a", "2 b"))
 		tk.MustQuery("select a,  name from t2 order by name").Check(testkit.Rows("2 a"))
+
+		// Test delete parent table in pessimistic txn
+		tk.MustExec("begin pessimistic")
+		tk.MustExec("insert into t2 (a, name) values (1, 'a');")
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			tk2.MustExec("begin pessimistic")
+			err := tk2.ExecToErr("delete from t1 where id = 1")
```

Comment:
Also test the non-fast path by `delete from t1 where id <= 1`.

Reply:
great, done.